pkg/metricscollector/opentelemetry.go

```diff
@@ -0,0 +1,255 @@
+package metricscollector
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"strconv"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	api "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/metric"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kedacore/keda/v2/version"
+)
+
+var otLog = logf.Log.WithName("prometheus_server")
+
+const meterName = "keda-open-telemetry-metrics"
+const defaultNamespace = "default"
+
+var (
+	meterProvider               *metric.MeterProvider
+	meter                       api.Meter
+	otErrorsTotalCounter        api.Int64Counter
+	otScalerErrorsCounter       api.Int64Counter
+	otScaledObjectErrorsCounter api.Int64Counter
+	otTriggerTotalsCounter      api.Int64UpDownCounter
+	otCrdTotalsCounter          api.Int64UpDownCounter
+)
+
+type OtelMetrics struct {
+}
+
+func NewOtelMetrics(options ...metric.Option) *OtelMetrics {
+	// create default options with env
+	if options == nil {
+		exporter, err := otlpmetrichttp.New(context.Background())
+		if err != nil {
+			fmt.Printf("Error:" + err.Error())
+			return nil
+		}
+		options = []metric.Option{metric.WithReader(metric.NewPeriodicReader(exporter))}
+	}
+
+	meterProvider = metric.NewMeterProvider(options...)
+	otel.SetMeterProvider(meterProvider)
+
+	meter = meterProvider.Meter(meterName)
+	initCounter()
+
+	otel := &OtelMetrics{}
+	otel.RecordBuildInfo()
+	return otel
+}
+
+func initCounter() {
+	var err error
+	msg := "create opentelemetry counter failed"
+
+	otErrorsTotalCounter, err = meter.Int64Counter("scaler.errors.total", api.WithDescription("Total number of errors for all scalers"))
+	if err != nil {
+		otLog.Error(err, msg)
+	}
+
+	otScalerErrorsCounter, err = meter.Int64Counter("scaler.errors", api.WithDescription("Number of scaler errors"))
```

Comment:
Is there any difference other than label/attribute presence in `scaler.errors.total` vs `scaler.errors` ? afaik the `sum()` function anywhere would still produce the same result, as long you don't filter it by any attribute.

Reply:
ref: https://github.com/kedacore/keda/pull/4860/files#diff-1b98f9b287bb8783cd25e2877e1897ecfca1e2a73c6852339679014d211c51ecR170-R172

The same applies to the Prometheus implementation. Got confused with that one for a sec.

Reply:
I think that is for easy analytic? Do you know the main reason of these metrics? @tomkerkhove @JorTurFer

Reply:
Yeah, we added them in the past to make the things easier for end users but it's true that probably it's worse for end users and they have to store another time series when Prometheus can just sum the values. IMO we can deprecate those repeated metrics if we documented it well, but we can't remove them directly and I also think that prometheus and otel should expose the same metrics.
Any thought @zroubalik @tomkerkhove ?

Reply:
I would deprecate them in Prometheus and if possible not even introduce in otel.

Reply:
@tomkerkhove ?

Reply:
Does sending the data in multiple ways make sense? Ingesting and processing telemetry means a cost, and we are generating multiple time series for things that end users can aggregate just in time if they need it.

Reply:
@tomkerkhove FYI^

Reply:
If we still push the raw `scaler.errors` metric and not remove it then I'm fine

Reply:
Yeah, the proposal is not losing any info but dropping aggregated time series which can be calculated on demand in the backend

Reply:
> If we still push the raw `scaler.errors` metric and not remove it then I'm fine

@SpiritZhou if we still have  the raw metric than you can drop the aggregated one